napari_clusters_plotter/_utilities.py

```diff
@@ -102,27 +103,50 @@ def update_properties_list(widget, exclude_list):
                     item.setSelected(True)
 
 
-def generate_cluster_image(label_image, predictionlist):
+def generate_cluster_image(layer, predictionlist):
     """
     Returns a label image where each label value corresponds
     to the cluster identity defined by the predictionlist.
     it is assumed that len(predictionlist) == max(label_image)
 
     Parameters
     ----------
-    label_image: ndarray or dask array
-        Label image used for cluster predictions
+    layer: napari.types.Layer
+        Layer used for cluster predictions
     predictionlist: array
         Array containing cluster identities for each label
     """
 
-    # reforming the prediction list, this is done to account
-    # for cluster labels that start at 0, conveniently hdbscan
-    # labelling starts at -1 for noise, removing these from
-    # the labels
-    predictionlist_new = np.array(predictionlist) + 1
-
-    return relabel(label_image, list(predictionlist_new)).astype("uint64")
+    if isinstance(layer, napari.layers.Labels):
+        label_image = layer.data
+        # reforming the prediction list this is done to account
+        # for cluster labels that start at 0, conveniently hdbscan
+        # labelling starts at -1 for noise, removing these from
+        # the labels
+        predictionlist_new = np.array(predictionlist) + 1
+        predictionlist_new = np.insert(predictionlist_new, 0, 0)
+
+        # loading data into gpu
+        clelist = cle.push(predictionlist_new)
+        gpu_labelimage = cle.push(label_image)
+
+        # generation of cluster label image
+        parametric_image = cle.replace_intensities(gpu_labelimage, clelist)
+        gpu_labelimage = None
+        clelist = None
+
+        # retrieving the gpu image
+        output = cle.pull(parametric_image).astype("uint32")
+        parametric_image = None
```

Comment:
I think [Relabel Image - change how the measurements list is reformed #34](https://github.com/haesleinhuepf/napari-skimage-regionprops/pull/34) in napari-skimage-regionprops should be merged, and then the following code changed to:
```suggestion
        output = relabel(label_image, list(predictionlist_new)).astype("uint32")
```

Reply:
Done! :-)